build/jobs/run-tests.yml

```diff
@@ -105,6 +97,67 @@ jobs:
       testSelector: testAssemblies
       testAssemblyVer2: |
         **\*${{ parameters.version }}.Tests.E2E*.dll
+      testFiltercriteria: "FullyQualifiedName~CosmosDb"
+      searchFolder: '$(System.ArtifactsDirectory)'
+      rerunFailedTests: true
+      rerunType: 'basedOnTestFailurePercentage'
+      rerunFailedThreshold: '95'
+      rerunMaxAttempts: 3
+
+- job: 'sqlE2eTests'
+  dependsOn: []
+  pool:
+    vmImage: $(WindowsVmImage)
+  steps:
+  - template: e2e-setup.yml
+    
+  - task: AzurePowerShell@4
+    displayName: 'Set Variables'
+    inputs:
+      azureSubscription: $(ConnectedServiceName)
+      azurePowerShellVersion: latestVersion
+      ScriptType: inlineScript
+      Inline: |
+        $secrets = Get-AzKeyVaultSecret -VaultName $(DeploymentEnvironmentName)-ts
+        
+        foreach($secret in $secrets)
+        {
+            $environmentVariableName = $secret.Name.Replace("--","_")
+
+            $secretValue = Get-AzKeyVaultSecret -VaultName $(DeploymentEnvironmentName)-ts -Name $secret.Name
+            Write-Host "##vso[task.setvariable variable=$($environmentVariableName)]$($secretValue.SecretValueText)"
+        }
+
+        $storageAccounts = Get-AzStorageAccount -ResourceGroupName $(DeploymentEnvironmentName)
+        foreach ($storageAccount in $storageAccounts) {
+            $accKey = Get-AzStorageAccountKey -ResourceGroupName $(DeploymentEnvironmentName) -Name $storageAccount.StorageAccountName | Where-Object {$_.KeyName -eq "key1"}
+
+            $storageSecretName = "$($storageAccount.StorageAccountName)_secret"
+            Write-Host "##vso[task.setvariable variable=$($storageSecretName)]$($accKey.Value)"
+        }
+
+        Write-Host "##vso[task.setvariable variable=Resource]$(TestEnvironmentUrl)"
+        
+        $secrets = Get-AzKeyVaultSecret -VaultName resolute-oss-tenant-info
+ 
+        foreach($secret in $secrets)
+        {
+            $environmentVariableName = $secret.Name.Replace("--","_")
+
+            $secretValue = Get-AzKeyVaultSecret -VaultName resolute-oss-tenant-info -Name $secret.Name
+            Write-Host "##vso[task.setvariable variable=$($environmentVariableName)]$($secretValue.SecretValueText)"
+        }
+        # ----------------------------------------
+
+        dotnet dev-certs https
+
+  - task: VSTest@2
+    displayName: 'Run E2E Tests'
+    inputs:
+      testSelector: testAssemblies
+      testAssemblyVer2: |
+        **\*${{ parameters.version }}.Tests.E2E*.dll
+      testFiltercriteria: "FullyQualifiedName~SqlServer"
```

Comment:
Another question for my understanding: why was `testFiltercriteria` added?

Reply:
Test filter criteria is what splits the tests into CosmosDB and SqlServer tests. This argument filters tests based on their names, and the test names have the name of the database included in them. The filter makes it so only the Cosmos tests run in one job and only the Sql tests run in the other job.

Reply:
Ah I see. Thank you for the explanation.